osu.Game/Screens/OnlinePlay/Multiplayer/MultiplayerMatchSubScreen.cs

```diff
@@ -258,6 +266,22 @@ private void onLoadRequested()
             readyClickOperation = null;
         }
 
+        private void settingsOverlayVisibilityChanged(ValueChangedEvent<Visibility> settingsOverlayVisibilityChangedEvent)
+        {
+            if (client.Room != null)
+            {
+                subScreenContainer.Show();
+                settingsOverlayVisibility.ValueChanged -= settingsOverlayVisibilityChanged;
+            }
+            else
+            {
+                if (settingsOverlayVisibilityChangedEvent.NewValue == Visibility.Visible)
+                    subScreenContainer.Hide();
+                else
+                    subScreenContainer.Show();
+            }
```

Comment:
I'm going to be honest, I'm not following this resolution. What is this trying to do? Can this be fixed just by making sure the sub screen container starts hidden, and showing it once the room has been created, without this logic?

If this is really necessary then I think this direly needs comments.

Reply:
> Can this be fixed just by making sure the sub screen container (or whichever part of the screen is showing there for a split second) starts hidden, and showing it once the room has been created, without this logic?

I've chosen to hide the sub screen container, which is what was showing for a split second, initially and then handle the visibility in this method mainly because I thought it would be more thorough for any cases that may not have been obvious to me.

Here are some comments explaining what this code is doing.
```csharp
if (client.Room != null)
{
    // If the room has been created show the sub screen container and unsubscribe from this event.
    subScreenContainer.Show();
    settingsOverlayVisibility.ValueChanged -= settingsOverlayVisibilityChanged;
}
else
{
    // The room hasn't been created yet
    if (settingsOverlayVisibilityChangedEvent.NewValue == Visibility.Visible)
        // If the initial settings window is visible then hide the sub screen container
        subScreenContainer.Hide();
    else
        // If the initial settings window is hidden then show the sub screen container
        subScreenContainer.Show();
}
```
The second else statement was mainly included in case there was a scenario where the initial settings window could be hidden without creating a room. If this isn't something that can happen then this method can be simplified to just the initial if statement.

Reply:
I don't believe it's possible to have both `client.Room` be null and the settings overlay hidden, and if it is, it's a bug. The room needs the initial settings to be created.

I think that this could be vastly simplified by hiding what needs to be hidden at the start, and then subscribing to `client.RoomUpdated`. If that event fires and `client.Room` is not null, then it should be safe to show the actual room screen contents.

Reply:
Okay, that does in fact simplify things then. I will work on subscribing to `client.RoomUpdated` with the `client.Room` check and see if this handles that. Thanks!